### ISSUE TITLE
vscode profile mode stuck at "Compiling ... for web"

### ISSUE THREAD

**Issue by robturtle** [flutter/devtools#3208]
## Issue
When launching with profile mode, chrome device, the app is actually loaded, but on vscode it's still showing "Compiling ... for web". If I try to open dart devtool from vscode, it will always reject me with "this debug session is not ready yet".

## Environment
VS Code
```
Version: 1.58.2
Commit: c3f126316369cd610563c75b1b1725e0679adfb3
Date: 2021-07-14T22:09:06.581Z (5 days ago)
Electron: 12.0.13
Chrome: 89.0.4389.128
Node.js: 14.16.0
V8: 8.9.255.25-electron.0
OS: Darwin x64 20.5.0
```
VS Code Flutter plugin version: `3.24.0`
VS Code Dart plugin version: `3.24.2`
launch.json
```json
{
  "version": "0.2.0",
  "configurations": [
    {
      "name": "flutter_app",
      "request": "launch",
      "type": "dart"
    },
    {
      "name": "flutter_app (profile mode)",
      "request": "launch",
      "type": "dart",
      "flutterMode": "profile"
    }
  ]
}
```

## Screenshot
![image](https://user-images.githubusercontent.com/3524125/126277877-2ab74ce1-789c-45f6-ae34-65b98f666599.png)

**Comment by breeze90** [comment]
Any update on it? Got the same problem.

**Comment by kenzieschmoll** [comment]
@DanTup

**Comment by DanTup** [comment]
@breeze90 I don't seem to be able to reproduce this. Can you confirm:

- the version of your Dart extension in VS Code
- the version if your Flutter SDK shown when you hover over the `{}` next to Dart/Flutter in the status bar in VS Code
- instructions on how you're launching the app

A screenshot of the full VS Code editor when it's suck at this step may also be useful.

Thanks!

**Comment by breeze90** [comment]
Dart v3.60.1
sdk: '>=2.19.0 <3.0.0'
video with launch attached

https://drive.google.com/file/d/1dUguKLNkgn78M8vuHkRpucKrX6T7KLiI/view?usp=sharing

**Comment by DanTup** [comment]
@breeze90 thanks - does this occur in a newly created counter app?

Could you capture a log for me (ideally using the counter app if it occurs there, so the project is closer to what I'm testing with):

- Run the [**Dart: Capture Debugging Logs** command](https://dartcode.org/docs/logging/#capture-debugging-logs) from the command palette (`F1`)
- Try to launch the app in profile mode and wait for the app to appear in Chrome
- Click **Cancel** on the logging notification to stop logging and open the log
- Ensure there's nothing sensitive in the log and attach here/in a gist

Thanks!

**Comment by breeze90** [comment]
yes looks like it does occur the same way.
[Dart-Code-Log-2023-02-01 20-27-53.txt](https://github.com/flutter/devtools/files/10960012/Dart-Code-Log-2023-02-01.20-27-53.txt)

I hope I did right with recording logs.

**Comment by DanTup** [comment]
@breeze90 you did, thanks!

I was able to reproduce this. I was using the new debug adapters (using the `dart.previewSdkDaps` setting) which doesn't have this problem. There are some outstanding issues on Stable that mean we haven't migrated to them yet.

I've filed https://github.com/Dart-Code/Dart-Code/issues/4437 to look at whether this is an easy fix in the old debug adapters.

I think the issue is just cosmetic though and shouldn't actually affect the debugging session.
